app/src/cnm/java/org/eyeseetea/malariacare/data/sync/importer/ConvertFromApiVisitor.java

```diff
@@ -0,0 +1,156 @@
+                    provinceOptions.add(optionProb);
+                    optionProb.save();
+                } else {
+                    optionProb = provinceOptions.get(provinceOptions.indexOf(optionProb));
+                }
+
+                if (orgUnitTree.getName_Dist_E() != null
+                        && !orgUnitTree.getName_Dist_E().isEmpty()) {
+                    OptionDB optionDist = new OptionDB(orgUnitTree.getName_Dist_E(),
+                            orgUnitTree.getName_Dist_E() + "", 0f, questionDB.getAnswerDB());
+                    optionDist.setId_parent_fk(optionProb.getId_option());
+                    if (!districtOptions.contains(optionDist)) {
+                        districtOptions.add(optionDist);
+                        optionDist.save();
+                    } else {
+                        optionDist = districtOptions.get(districtOptions.indexOf(optionDist));
+                    }
+
+                    if (orgUnitTree.getName_Comm_E() != null
+                            && !orgUnitTree.getName_Comm_E().isEmpty()) {
+                        OptionDB optionComm = new OptionDB(orgUnitTree.getName_Comm_E(),
+                                orgUnitTree.getName_Comm_E() + "", 0f,
+                                questionDB.getAnswerDB());
+                        optionComm.setId_parent_fk(optionDist.getId_option());
+                        if (!communeOptions.contains(optionComm)) {
+                            communeOptions.add(optionComm);
+                            optionComm.save();
+                        } else {
+                            optionComm = communeOptions.get(communeOptions.indexOf(optionComm));
+                        }
+
+                        if (orgUnitTree.getName_Vill_E() != null
+                                && !orgUnitTree.getName_Vill_E().isEmpty()) {
+                            OptionDB optionVill = new OptionDB(orgUnitTree.getName_Vill_E(),
+                                    orgUnitTree.getName_Vill_E() + "", 0f,
+                                    questionDB.getAnswerDB());
+                            optionVill.setId_parent_fk(optionComm.getId_option());
+                            OptionAttributeDB optionAttributeDB = new OptionAttributeDB();
+                            optionAttributeDB.setPath(
+                                    "[" + orgUnitTree.getLat() + "," + orgUnitTree.getLng()
+                                            + "]");
+                            villageOptionAttributes.add(optionAttributeDB);
+                            optionVill.setOptionAttributeDB(optionAttributeDB);
+                            villageOptions.add(optionVill);
+                        }
+                    }
+                }
+            }
+        }
+        Log.d("SAVING LIST", "saving optrionstree list");
+        saveBatch(villageOptionAttributes);
+        for (Model villageOption : villageOptions) {
```

Comment:
Please review this loops because has no sense, for every option is assign every attribute. It is 100 million loops and performance is affected, I think that creates a map and search by keymap can to improve performance